lightning/src/chain/mod.rs

```diff
@@ -129,9 +129,34 @@ pub trait Filter: Send + Sync {
 	/// a spending condition.
 	fn register_tx(&self, txid: &Txid, script_pubkey: &Script);
 
-	/// Registers interest in spends of a transaction output identified by `outpoint` having
-	/// `script_pubkey` as the spending condition.
-	fn register_output(&self, outpoint: &OutPoint, script_pubkey: &Script);
+	/// Registers interest in spends of a transaction output.
+	///
+	/// Optionally, when `output.block_hash` is set, should return any transaction spending the
+	/// output that is found in the corresponding block at the given index.
+	///
+	/// This return value is useful for Electrum clients in order to supply in-block descendant
+	/// transactions which otherwise were not included. This is not necessary for other clients if
+	/// such descendant transactions were already included (e.g., when a BIP 157 client provides the
+	/// full block).
+	fn register_output(&self, output: WatchedOutput) -> Option<(usize, Transaction)>;
+}
+
+/// A transaction output watched by a [`ChannelMonitor`] for spends on-chain.
+///
+/// Used to convey to a [`Filter`] an output identified by `outpoint` having `script_pubkey` as the
+/// spending condition. If `block_hash` is `Some`, this indicates the output was spent in the
+/// corresponding block.
```

Comment:
FYI, I think this last sentence needs to be rephrased. The presence of a block hash should indicate it *may have been* spent in the corresponding block. Open to an alternative rephrasing.

Reply:
I'm good with "may have been spent" ? I don't think we should leak more protocol information like when the descendant *can't* have been spent in same block due to relative-timelock usage.